lib/ClangImporter/ClangImporter.cpp

```diff
@@ -6364,6 +6364,11 @@ TinyPtrVector<ValueDecl *> ClangRecordMemberLookup::evaluate(
         continue;
 
       auto *baseRecord = baseType->getAs<clang::RecordType>()->getDecl();
+
+      if (isSymbolicCircularBase(cxxRecord, baseRecord))
```

Comment:
My last force push updated the test case to cover one more case (a class that inherits from a circularly inheriting class), and fixed this code to check for the right base. It used to check the C++ class of `inheritingDecl`, but now it checks the C++ class of `recordDecl`, where:

```
template <typename T> class Circular;
template <> class Circular<void> { ... };
template <typename T> class Circular<T> : Circular<void> { ... };

typedef Circular<SomeType> RecordDecl;
class InheritingDecl : RecordDecl { ... };
```

Neither check is complete, but heuristically checking `recordDecl` is probably better (at least that's what matches the circular inheritance case in `std::exception`).